secrets/secrets_test.go

```diff
@@ -412,3 +416,66 @@ func TestSecrets(t *testing.T) {
 	}
 
 }
+
+// Check if all the rules that exist in "gitleaks" are included in our list of rules (In secret.go file)
+func TestAllGitleaksRulesAreUsed(t *testing.T) {
+
+	// Import the rules from "gitleaks"
+	response, err := http.Get("https://raw.githubusercontent.com/gitleaks/gitleaks/master/cmd/generate/config/main.go")
+	if err != nil {
+		fmt.Printf("Failed to fetch remote file: %v\n", err)
+		return
+	}
+	defer response.Body.Close()
+
+	content, err := io.ReadAll(response.Body)
+	if err != nil {
+		fmt.Printf("Failed to read remote file content: %v\n", err)
+		return
+	}
+
+	reGitleaks := regexp.MustCompile(`configRules\s*=\s*append\(configRules,\s*rules\.([a-zA-Z0-9_]+)\(`)
+	matchesGitleaks := reGitleaks.FindAllStringSubmatch(string(content), -1)
+
+	var gitleaksRules []string
+
+	for _, match := range matchesGitleaks {
+		gitleaksRules = append(gitleaksRules, match[1])
+	}
+
+	//Import the rules from our project "2ms"
+
+	localfile, err := os.ReadFile("secrets.go")
+	if err != nil {
+		t.Errorf("Failed to read local file content: %v", err)
+	}
+
+	reLocal := regexp.MustCompile(`allRules\s*=\s*append\(allRules,\s*Rule{Rule:\s*\*rules\.([a-zA-Z0-9_]+)\(\),`)
+	matchLocal := reLocal.FindAllStringSubmatch(string(localfile), -1)
+
+	localRulesMap := make(map[string]bool)
+
+	for _, match := range matchLocal {
+		localRulesMap[match[1]] = true
+	}
```

Comment:
The logic of taking a content and regexing it into a list is duplicated. Do you think you can unifiy it?

Reply:
I want to move about all the rules in "gitleaks" and check about each one if it exists or not in our rules.
So, In the case of the rules in the "gitleaks" you are right, and I will correct it.
But in the case of our rules, I think is preferable to insert our rules into a "map" for a good run time when we search each one.

Do you agree with me?